rwgame/debug/TcpSocket.cpp

```diff
@@ -0,0 +1,67 @@
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <arpa/inet.h>
+#include <unistd.h>
+#include <vector>
+
+#include "TcpSocket.hpp"
+
+
+TcpSocket::TcpSocket() :
+	sock(-1), addr()
+{
+	sock = socket(AF_INET, SOCK_STREAM, 0);
+}
+
+
+void TcpSocket::bind(short port)
+{
+	addr.sin_family = AF_INET;
+	addr.sin_addr.s_addr = htonl(INADDR_ANY);
+	addr.sin_port = htons(port);
+
+	::bind(sock, reinterpret_cast<sockaddr*>(&addr), sizeof(addr));
+}
+
+
+void TcpSocket::listen(TcpSocket& client)
+{
+	::listen(sock, 1);
+
+	socklen_t len = sizeof(client.addr);
+	client.sock = accept(sock, reinterpret_cast<sockaddr*>(&client.addr), &len);
+}
+
+
+void TcpSocket::recv(std::string& out, size_t len)
+{
+	std::vector<char> buffer(len);
+	read(sock, buffer.data(), len);
+
+	out = { buffer.begin(), buffer.end() };
+}
+
+
+void TcpSocket::send(const std::string& str)
+{
+	write(sock, str.c_str(), str.length());
+}
+
+
+void TcpSocket::disconnect()
+{
+	close(sock);
```

Comment:
`/// @todo must check if socket exists or check for errors in general`

also you might want to set `sock = -1` again or something?